### PR TITLE
Migrates tests from Boost.Test to Catch framework.

### PR DESCRIPTION
**Pull request by lexicalunit**
Move away from Boost.Test; use the much nicer Catch instead.

### CODE REVIEW

**Comment by pepr** [comment]
You have mentioned BDD... I have created a tool -- see https://github.com/pepr/BDDtool . Written in Python. I did not finish the second part of my goals -- it requires more work.

I was just learning what it really means, why other people created the tools like cucumber. In my opinion, some of the other tools are too much clever. My goal was just to convert the test description in `features/` to the Catch skeletons in `tests/`. I did not used it for the "real" development; anyway, it seems to be nice approach. Similarly to TDD, it forces you to think more before you really implement.

The second goal (not implemented) was to synchronize changes from the test files backward to the `.feature` files.

The next goal was to rewrite the tool from Python to C++ -- I guess it would be much more acceptable by others.

If you find it interesting, feel free to ask. If you have friends who may be interested, tell them. Otherwise, the project is rather dormant.

Have a nice day!

**Comment by mloskot** [comment]
:+1:

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +6,9 @@ include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src)
 foreach(example ${examples})
 	add_executable(example_${example} ${example}.cpp)
 	if (NANODBC_STATIC)
-		target_link_libraries(example_${example} nanodbc ${Boost_LIBRARIES} ${ODBC_LIBRARIES})
```
Comment (mloskot) [inline examples/CMakeLists.txt]: I have missed to review the details of this PR.
This actually drops the fixes in 37c3641a6de17a71442d224a889882cb32937d12 and 631e4496712ea17db2e83aa7a2b8df3c3c36ca08 leading to linker errors on Windows.
Use of `ODBC_LIBRARIES` is required.

**Comment by mloskot** [comment]
@lexicalunit Before releasing next nanodbc version, you may want to wait until Clara has been fixed (see my PR I submitted to Clara and auto-linked above).

**Comment by lexicalunit** [comment]
Sounds good! :+1:

**Pull request merged**